### DISCUSSION TITLE
`mitmdump` with `--certs ` doesnt work for iOS

### DISCUSSION THREAD

**Discussion by zkcrescent** [mitmproxy/mitmproxy#6509] [General]
#### Problem Description
directly executing `mitmdump` works well, all request can be proxyed,  but when I run `mitmdump --certs '~/.mitmproxy/mitmproxy-ca-cert.pem'` shown two errors:
- Client TLS handshake failed. Client and mitmproxy cannot agree on a TLS version to use. You may need to adjust mitmproxy's tls_version_client_min option.
- Client TLS handshake failed. The client may not trust the proxy's certificate for  xxx.com `(domains like captive.apple.com)` (OpenSSL Error([('SSL routines', '', 'sslv3 alert illegal parameter')]))
#### Steps to reproduce the behavior:
1. install `~/.mitmproxy/mitmproxy-ca-cert.pem` on iOS
2. trust `mitmproxy` on iOS
3. run `mitmdump` on machine
4.  all pages on iOS work well 
5. rerun  `mitmdump --certs '~/.mitmproxy/mitmproxy-ca-cert.pem'` on machine
6. all pages on iOS cant work

#### System Information

Mitmproxy: 10.1.5
Python:    3.11.5
OpenSSL:   OpenSSL 3.1.4 24 Oct 2023
Platform:  macOS-13.5.1-arm64-arm-64bit

**Comment by mhils** [comment]
You can use `--certs` to specify a specific _leaf_ (server) certificate. If you want to specify a custom certificate authority instead, see https://docs.mitmproxy.org/stable/concepts-certificates/#using-a-custom-certificate-authority :)
